daffodil-test/src/test/java/org/goodudfs/example/StringFunctions/StringFunctionsProvider.java

```diff
@@ -0,0 +1,25 @@
+package org.goodudfs.example.StringFunctions;
+
+import org.apache.daffodil.udf.*;
+
+public class StringFunctionsProvider extends UDFunctionProvider {
+	public StringFunctionsProvider() {
+		super.setFunctionClasses( new Class<?>[] { Replace.class, Compare.class } );
+	}
+
+	public Object lookupFunctionClass(String namespace, String name) {
+		Object functionClass = null;
+
+		String nn = String.join("_", namespace, name);
+
+		switch (nn) {
+		case "com.ext.UDFunction.StringFunctions_replace":
+			functionClass = new Replace();
+			break;
+		case "com.ext.UDFunction.StringFunctions_compare":
+			functionClass = new Compare();
+			break;
+		}
+		return functionClass;
```

Comment:
The naming here is a bit confusing. It sounds like you are returning a the Class<?> object corresponding to the function, but you are actually returning an instantiated object.

I suggest renaming to functionObject here and in the method name.

Reply:
Ahh yes, this was definitely misnamed and has been updated to actually do the function class and evaluate method checks. We now return the functionClassObject, evaluateMethod, paramTypes and retType. So it's called lookupUDFCallingInfo. Any naming suggestions welcome!